linera-views/src/dynamo_db.rs

```diff
@@ -350,44 +375,61 @@ impl DynamoDbBatch {
         let delete_count = self.0.deletions.len();
         let insert_count = self.0.insertions.len();
         let total_count = delete_count + insert_count;
-        let mut curr_size = 0;
+        let mut value_size = 2;
+        let mut transact_size = 0;
         let mut curr_len = 0;
         let mut deletions = Vec::new();
         let mut insertions = Vec::new();
         let mut block_count = 0;
+        let mut transacts = Vec::new();
         for i in 0..total_count {
             curr_len += 1;
             if i < delete_count {
                 let delete = &self.0.deletions[i];
-                curr_size += delete.len();
+                value_size += delete.len() + 1;
                 deletions.push(delete.to_vec());
```

Comment:
Since we own `self`, I think there's a way to do this without cloning, and without duplicating the `if … < delete_count` distinction. But I can make a suggestion as a PR later.

Reply:
…something like this: https://github.com/MathieuDutSik/linera-protocol_second/pull/1

I even wonder whether it should be two or even three nested loops—the inner ones being for the transactions, and then for the individual deletes and inserts. But maybe that would duplicate the conditions for stopping.

Reply:
That is indeed nicer.